amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/DiscriminatedFieldAccessor.java

```diff
@@ -43,18 +46,23 @@ public DiscriminatedFieldAccessor(FieldAccessor target, String discriminatorFiel
     @Override
     public Object get(Object targetObject) {
         Collection collection = getWrappedCollection(targetObject);
+        String prefix = "" + TLSUtils.getRequest().getAttribute(WORKSPACE_PREFIX);
         if (multipleValues) {
             List<Object> filteredItems = new ArrayList<>();
             for (Object item : collection) {
-                if (getDiscriminationValue(item).equals(discriminatorValue)) {
+                // AMPOFFLINE-1528
+                if (getDiscriminationValue(item).equals(discriminatorValue)
+                        || getDiscriminationValue(item).equals(prefix + discriminatorValue)) {
```

Comment:
I think we need to check with prefix and if not use without not both at a time as we do in online. but I might be wrong, lets discuss

Reply:
We need both.